aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIO.java

```diff
@@ -298,8 +303,13 @@ private List<String> deleteBatch(String bucket, Collection<String> keysToDelete)
   @Override
   public Iterable<FileInfo> listPrefix(String prefix) {
     S3URI s3uri = new S3URI(prefix, s3FileIOProperties.bucketToAccessPointMapping());
+    String suffix = "";
```

Comment:
I think there is a correctness issue here, because people that list `s3://bucket/some-prefix` does not necessarily mean `some-prefix` is a directory name, and doing so would create wrong behavior. This includes directory bucket users, maybe the user is switching from general purpose bucket to directory bucket, and they have been listing prefix not a directory, the expected behavior is for it to fail, rather than listing with a / which returns empty result.

But I agree it is probably more convenient to do this trick for directory buckets. Maybe we can introduce a flag here, something like `s3.directory-bucket.list-prefix-as-directory` which defaults to true, and only do this trick when the flag is on, so the correct behavior is preserved for people who want to have that. What do you think?